profiles/models.py

```diff
@@ -21,12 +23,40 @@ def __repr__(self) -> str:
         return '<OrcidToken for %r>' % self.orcid
 
 
+class Name(object):
+    def __init__(self, given_names: str, family_name: str = None) -> None:
+        self.given_names = given_names
+        self.family_name = family_name
+
+    def __composite_values__(self) -> Iterable[str]:
+        return self.given_names, self.family_name
+
+    def __str__(self) -> str:
+        if self.family_name is None:
+            return self.given_names
+
+        return '%s %s' % (self.given_names, self.family_name)
+
+    def __repr__(self) -> str:
+        return '<Name %r %r>' % (self.given_names, self.family_name)
+
+    def __eq__(self, other: Any) -> bool:
+        return isinstance(other, Name) and \
+               other.given_names == self.given_names and \
+               other.family_name == self.family_name
+
+    def __ne__(self, other: Any) -> bool:
+        return not self.__eq__(other)
+
+
 class Profile(db.Model):
     id = db.Column(db.String(ID_LENGTH), primary_key=True)
-    name = db.Column(db.Text(), nullable=False)
+    name = composite(Name, '_given_names', '_family_name')
+    _given_names = db.Column(db.Text(), name='given_names', nullable=False)
+    _family_name = db.Column(db.Text(), name='family_name')
```

Comment:
do we need to store parts of the name separately? They may not be available when using non-ORCID data sources

Reply:
also, we may need to store `index` values like https://github.com/elifesciences/api-raml/pull/192/files#diff-6cbf65dc93ac1e6649327ba4a25bc550R15 directly as a computed field to allow for (fast) ordering (or what that field is used for?)

Reply:
It is true we lose information, but it can be retrieved from the API and this is our data model's, not ORCID's

Reply:
Hmm, had thought that sorting by family then given names would work, but that would actually produce different results (when there is no family name).

In the future I envisaged a requirement for an 'address' name (ie Hello Chris in an email), but we can always cross that bridge when it comes.